client/post-editor/post-editor.jsx

```diff
@@ -574,10 +578,16 @@ export const PostEditor = React.createClass( {
 		this.setState( { isSaving: true } );
 	},
 
-	onPreview: function( event ) {
+	onPreview: function( action, event ) {
 		var status = 'draft',
 			previewPost;
 
+		if ( this.state.previewAction !== action ) {
+			this.setState( {
+				previewAction: action
+			} );
+		}
```

Comment:
>In that scenario, `this.state.previewAction` would still be set to `preview` so we're avoiding an extra `setState` call.

Hmm, kinda goes to highlight that letting the state be long-lived beyond the initial view/preview intent is perhaps a bit fragile. I don't really have a good alternative to suggest, but a little unnerving.